salt/states/x509.py

```diff
@@ -308,6 +308,9 @@ def private_key_managed(name,
         file_args['contents'] = __salt__['x509.create_private_key'](
             text=True, bits=bits, passphrase=passphrase, cipher=cipher, verbose=verbose)
 
+    # Ensure the key contents are a string before passing it along
+    file_args['contents'] = salt.utils.stringutils.to_str(file_args['contents'])
```

Comment:
Is there a specific reason (like maybe x509 not working with unicode strings in PY2) that we're using `to_str()` here?

Reply:
Exactly that. It's expecting it to be a string and not bytes.

Reply:
:+1: